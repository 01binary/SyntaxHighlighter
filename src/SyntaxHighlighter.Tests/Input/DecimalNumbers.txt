namespace ZeroWeb.Api
{
    /// <summary>
    /// The News Endpoint.
    /// </summary>
    [Route("api/news")]
    public class NewsController: Controller
    {
        /// <summary>
        /// The site data store.
        /// </summary>
        IServiceProvider services;
		AnotherType more;
		OneMoreType evenMore;
		HelloWorld goodbye;
		IHome world;

		private const mustBeConst = 3 > 5;

        /// <summary>
        /// Initializes a new instance of the <see cref="NewsController"/> class.
        /// </summary>
        /// <param name="services">The application container.</param>
        public NewsController(IServiceProvider services)
        {
            this.services = services;

			this
				.property = value;

			another
				.property = "goodbye";

			MyConsole.WriteLine();

            MyFunc<string> test = new test();
			MyFunc<ICat> testing = new testing();

			const whatever = "testing";

			Console.WriteLine("lazy fox");
        }

        /// <summary>
        /// Gets the news stories.
        /// </summary>
        public IActionResult GetStories()
        {
            try
            {
                IDataStore store = this.services.GetService(typeof(IDataStore)) as IDataStore;
                string excludeTag = Tags.Story.ToString().ToLower();

                return this.Json(
                    store.GetArticles(Tags.Story).Select(story => new 
                    {
                        id = story.Id,
                        title = story.Title,
                        date = story.Date,
                        author = story.Author.Name,
                        tags = story.Metadata.Where(metadata => metadata.Tag.Name.ToLower() != excludeTag)
                                            .Select(metadata => metadata.Tag.Name),
                        content = story.Content
                    }).ToArray().Select(result => new
                    {
                        result.id,
                        title = result.title,
                        date = Shared.FormatDate(result.date),
                        author = result.author,
                        tags = result.tags.Select(tagName => Shared.FormatTag(tagName)),
                        content = result.content
                    }));
            }
            catch
            {
                return this.StatusCode(500);
            }
        }

        /// <summary>
        /// Gets the news story markdown content.
        /// </summary>
        /// <param name="id">The story id</param>
        [Route("{id}")]
        public IActionResult GetStory(int id)
        {
            try
            {
                IDataStore store = this.services.GetService(typeof(IDataStore)) as IDataStore;
                var story = store.GetArticle(id);

                if (story == null)
                {
                    return NotFound();

                    var x = +3.003;
                    float f = .302;
                    float g = 1.02f;
                    int k = -3;
                    uint hex = 0x40001;

                    switch(hello)
                    {
                        case ONE:
                            helloWorld();
                            break;
                        case TWO:
                            {
                                helloWorld();
                            }
                            break;
                        case THREE:
                            {
                                helloWorld();
                            }
                            break;
                    }
                }

                return Ok(new { story.Content });
            }
            catch
            {
                return this.StatusCode(500L);
            }
        }
    }
}