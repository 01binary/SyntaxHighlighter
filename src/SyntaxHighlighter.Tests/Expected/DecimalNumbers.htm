<!DOCTYPE html>
<html lang="en">
<head>
    <title>Code Highlight Preview</title>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width">
    <link rel="stylesheet" type="text/css" href="../Template/Styles.css">
</head>
<body>
<pre><span class="pl-k">namespace</span> <span class="pl-e">ZeroWeb</span>.<span class="pl-e">Api</span>
{
    <span class="pl-c">/// &lt;<span class="pl-ent">summary</span>&gt;</span>
    <span class="pl-c">/// The News Endpoint.</span>
    <span class="pl-c">/// &lt;/<span class="pl-ent">summary</span>&gt;</span>
    [<span class="pl-e">Route</span>(<span class="pl-s">"api/news"</span>)]
    <span class="pl-k">public</span> <span class="pl-k">class</span> <span class="pl-e">NewsController</span>: <span class="pl-en">Controller</span>
    {
        <span class="pl-c">/// &lt;<span class="pl-ent">summary</span>&gt;</span>
        <span class="pl-c">/// The site data store.</span>
        <span class="pl-c">/// &lt;/<span class="pl-ent">summary</span>&gt;</span>
        <span class="pl-en">IServiceProvider</span> <span class="pl-v">services</span>;
		<span class="pl-en">AnotherType</span> <span class="pl-v">more</span>;
		<span class="pl-en">OneMoreType</span> <span class="pl-v">evenMore</span>;
		<span class="pl-en">HelloWorld</span> <span class="pl-v">goodbye</span>;
		<span class="pl-en">IHome</span> <span class="pl-v">world</span>;

		<span class="pl-k">private</span> <span class="pl-k">const</span> <span class="pl-c1">mustBeConst</span> = <span class="pl-c1">3</span> > <span class="pl-c1">5</span>;

        <span class="pl-c">/// &lt;<span class="pl-ent">summary</span>&gt;</span>
        <span class="pl-c">/// Initializes a new instance of the &lt;<span class="pl-ent">see</span> <span class="pl-e">cref</span>=<span class="pl-s">"NewsController"</span>/&gt; class.</span>
        <span class="pl-c">/// &lt;/<span class="pl-ent">summary</span>&gt;</span>
        <span class="pl-c">/// &lt;<span class="pl-ent">param</span> <span class="pl-e">name</span>=<span class="pl-s">"services"</span>&gt;The application container.&lt;/<span class="pl-ent">param</span>&gt;</span>
        <span class="pl-k">public</span> <span class="pl-e">NewsController</span>(<span class="pl-en">IServiceProvider</span> <span class="pl-v">services</span>)
        {
            <span class="pl-c1">this</span>.<span class="pl-v">services</span> = <span class="pl-v">services</span>;

			<span class="pl-c1">this</span>
				.<span class="pl-v">property</span> = <span class="pl-v">value</span>;

			<span class="pl-v">another</span>
				.<span class="pl-e">property</span> = <span class="pl-s">"goodbye"</span>;

			<span class="pl-v">MyConsole</span>.<span class="pl-e">WriteLine</span>();

            <span class="pl-en">MyFunc</span><<span class="pl-k">string</span>> <span class="pl-v">test</span> = <span class="pl-k">new</span> <span class="pl-e">test</span>();
			<span class="pl-en">MyFunc</span><<span class="pl-en">ICat</span>> <span class="pl-v">testing</span> = <span class="pl-k">new</span> <span class="pl-e">testing</span>();

			<span class="pl-k">const</span> <span class="pl-c1">whatever</span> = <span class="pl-s">"testing"</span>;

			<span class="pl-c1">Console</span>.<span class="pl-e">WriteLine</span>(<span class="pl-s">"lazy fox"</span>);
        }

        <span class="pl-c">/// &lt;<span class="pl-ent">summary</span>&gt;</span>
        <span class="pl-c">/// Gets the news stories.</span>
        <span class="pl-c">/// &lt;/<span class="pl-ent">summary</span>&gt;</span>
        <span class="pl-k">public</span> <span class="pl-en">IActionResult</span> <span class="pl-e">GetStories</span>()
        {
            <span class="pl-k">try</span>
            {
                <span class="pl-en">IDataStore</span> <span class="pl-v">store</span> = <span class="pl-c1">this</span>.<span class="pl-v">services</span>.<span class="pl-e">GetService</span>(<span class="pl-k">typeof</span>(<span class="pl-en">IDataStore</span>)) <span class="pl-k">as</span> <span class="pl-en">IDataStore</span>;
                <span class="pl-k">string</span> <span class="pl-v">excludeTag</span> = <span class="pl-v">Tags</span>.<span class="pl-e">Story</span>.<span class="pl-e">ToString</span>().<span class="pl-e">ToLower</span>();

                <span class="pl-k">return</span> <span class="pl-c1">this</span>.<span class="pl-e">Json</span>(
                    <span class="pl-v">store</span>.<span class="pl-e">GetArticles</span>(<span class="pl-v">Tags</span>.<span class="pl-e">Story</span>).<span class="pl-e">Select</span>(<span class="pl-v">story</span> => <span class="pl-k">new</span> 
                    {
                        <span class="pl-v">id</span> = <span class="pl-v">story</span>.<span class="pl-e">Id</span>,
                        <span class="pl-v">title</span> = <span class="pl-v">story</span>.<span class="pl-e">Title</span>,
                        <span class="pl-v">date</span> = <span class="pl-v">story</span>.<span class="pl-e">Date</span>,
                        <span class="pl-v">author</span> = <span class="pl-v">story</span>.<span class="pl-e">Author</span>.<span class="pl-e">Name</span>,
                        <span class="pl-v">tags</span> = <span class="pl-v">story</span>.<span class="pl-e">Metadata</span>.<span class="pl-e">Where</span>(<span class="pl-v">metadata</span> => <span class="pl-v">metadata</span>.<span class="pl-e">Tag</span>.<span class="pl-e">Name</span>.<span class="pl-e">ToLower</span>() != <span class="pl-v">excludeTag</span>)
                                            .<span class="pl-e">Select</span>(<span class="pl-v">metadata</span> => <span class="pl-v">metadata</span>.<span class="pl-e">Tag</span>.<span class="pl-e">Name</span>),
                        <span class="pl-v">content</span> = <span class="pl-v">story</span>.<span class="pl-e">Content</span>
                    }).<span class="pl-e">ToArray</span>().<span class="pl-e">Select</span>(<span class="pl-v">result</span> => <span class="pl-k">new</span>
                    {
                        <span class="pl-v">result</span>.<span class="pl-e">id</span>,
                        <span class="pl-v">title</span> = <span class="pl-v">result</span>.<span class="pl-e">title</span>,
                        <span class="pl-v">date</span> = <span class="pl-v">Shared</span>.<span class="pl-e">FormatDate</span>(<span class="pl-v">result</span>.<span class="pl-e">date</span>),
                        <span class="pl-v">author</span> = <span class="pl-v">result</span>.<span class="pl-e">author</span>,
                        <span class="pl-v">tags</span> = <span class="pl-v">result</span>.<span class="pl-e">tags</span>.<span class="pl-e">Select</span>(<span class="pl-v">tagName</span> => <span class="pl-v">Shared</span>.<span class="pl-e">FormatTag</span>(<span class="pl-v">tagName</span>)),
                        <span class="pl-v">content</span> = <span class="pl-v">result</span>.<span class="pl-e">content</span>
                    }));
            }
            <span class="pl-k">catch</span>
            {
                <span class="pl-k">return</span> <span class="pl-c1">this</span>.<span class="pl-e">StatusCode</span>(<span class="pl-c1">500</span>);
            }
        }

        <span class="pl-c">/// &lt;<span class="pl-ent">summary</span>&gt;</span>
        <span class="pl-c">/// Gets the news story markdown content.</span>
        <span class="pl-c">/// &lt;/<span class="pl-ent">summary</span>&gt;</span>
        <span class="pl-c">/// &lt;<span class="pl-ent">param</span> <span class="pl-e">name</span>=<span class="pl-s">"id"</span>&gt;The story id&lt;/<span class="pl-ent">param</span>&gt;</span>
        [<span class="pl-e">Route</span>(<span class="pl-s">"{id}"</span>)]
        <span class="pl-k">public</span> <span class="pl-en">IActionResult</span> <span class="pl-e">GetStory</span>(<span class="pl-k">int</span> <span class="pl-v">id</span>)
        {
            <span class="pl-k">try</span>
            {
                <span class="pl-en">IDataStore</span> <span class="pl-v">store</span> = <span class="pl-c1">this</span>.<span class="pl-v">services</span>.<span class="pl-e">GetService</span>(<span class="pl-k">typeof</span>(<span class="pl-en">IDataStore</span>)) <span class="pl-k">as</span> <span class="pl-en">IDataStore</span>;
                <span class="pl-k">var</span> <span class="pl-v">story</span> = <span class="pl-v">store</span>.<span class="pl-e">GetArticle</span>(<span class="pl-v">id</span>);

                <span class="pl-k">if</span> (<span class="pl-v">story</span> == <span class="pl-c1">null</span>)
                {
                    <span class="pl-k">return</span> <span class="pl-e">NotFound</span>();

                    <span class="pl-k">var</span> <span class="pl-v">x</span> = <span class="pl-c1">+3.003</span>;
                    <span class="pl-k">float</span> <span class="pl-v">f</span> = <span class="pl-c1">.302</span>;
                    <span class="pl-k">float</span> <span class="pl-v">g</span> = <span class="pl-c1">1.02f</span>;
                    <span class="pl-k">int</span> <span class="pl-v">k</span> = <span class="pl-c1">-3</span>;
                    <span class="pl-k">uint</span> <span class="pl-v">hex</span> = <span class="pl-c1">0x40001</span>;

                    <span class="pl-k">switch</span>(<span class="pl-v">hello</span>)
                    {
                        <span class="pl-k">case</span> <span class="pl-c1">ONE</span>:
                            <span class="pl-e">helloWorld</span>();
                            <span class="pl-k">break</span>;
                        <span class="pl-k">case</span> <span class="pl-c1">TWO</span>:
                            {
                                <span class="pl-e">helloWorld</span>();
                            }
                            <span class="pl-k">break</span>;
                        <span class="pl-k">case</span> <span class="pl-c1">THREE</span>:
                            {
                                <span class="pl-e">helloWorld</span>();
                            }
                            <span class="pl-k">break</span>;
                    }
                }

                <span class="pl-k">return</span> <span class="pl-e">Ok</span>(<span class="pl-k">new</span> { <span class="pl-v">story</span>.<span class="pl-e">Content</span> });
            }
            <span class="pl-k">catch</span>
            {
                <span class="pl-k">return</span> <span class="pl-c1">this</span>.<span class="pl-e">StatusCode</span>(<span class="pl-c1">500L</span>);
            }
        }
    }
}</pre>
</body>
</html>